src/langpacks.py

```diff
@@ -0,0 +1,332 @@
+    "python3-apt",
+    "dput",
+    "git",
+    "devscripts",
+    "lintian",
+]
+
+HOME = Path("~ubuntu").expanduser()
+REPO_LOCATION = HOME / "langpack-o-matic"
+
+
+class Langpacks:
+    """Represent a langpacks instance in the workload."""
+
+    def __init__(self):
+        logger.debug("Langpacks class init")
+
+    def setup_crontab(self):
+        """Configure the crontab for the service."""
+        try:
+            run(
+                [
+                    "su",
+                    "-c",
+                    "crontab src/crontab",
+                    "ubuntu",
+                ],
+                check=True,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+            )
+            logger.debug("Crontab configured.")
+            return
+        except CalledProcessError as e:
+            logger.debug(f"Installation of the crontab failed: {e.stdout}")
+            raise
+
+    def install(self):
+        """Install the langpack builder environment."""
+        # Install the deb packages needed for the service
+        try:
+            apt.update()
+            logger.debug("Apt index refreshed.")
+        except CalledProcessError as e:
+            logger.error(f"failed to update package cache: {e}")
+            raise
+
+        for p in PACKAGES:
+            try:
+                apt.add_package(p)
+                logger.debug(f"Package {p} installed")
+            except PackageNotFoundError:
+                logger.error(f"failed to find package {p} in package cache")
+                raise
+            except PackageError as e:
+                logger.error(f"failed to install {p}: {e}")
+                raise
+
+        # Clone the langpack-o-matic repo
+        try:
+            run(
+                [
+                    "sudo",
+                    "-u",
+                    "ubuntu",
+                    "git",
+                    "clone",
+                    "-b",
+                    "master",
+                    "https://git.launchpad.net/langpack-o-matic",
+                    REPO_LOCATION,
+                ],
+                check=True,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+            )
+            logger.debug("Langpack-o-matic vcs cloned.")
+        except CalledProcessError as e:
+            logger.debug(f"Git clone of the code failed: {e.stdout}")
+            raise
+
+    def update_checkout(self):
+        """Update the langpack-o-matic checkout."""
+        # Pull Vcs updates
+        try:
+            run(
+                [
+                    "sudo",
+                    "-u",
+                    "ubuntu",
+                    "git",
+                    "-C",
+                    REPO_LOCATION,
+                    "pull",
+                ],
+                check=True,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+            )
```

Comment:
This pattern of calling `run` with `sudo -u ubuntu` and catching the `CalledProcessError` is common enough that some sort of helper might make the code a little more readable, but not mandatory to progress.

Reply:
I moved away from changing user for the job which resolves that point